### PR TITLE
Update custom permission rule documentation

### PR DESCRIPTION
**Pull request by stanley-nicholl**
## Hey, I just made a Pull Request!

Updated the custom permission rule documentation to fix several inaccuracies and improve the readability of the document. I have been very deliberate to establish descriptive variable names to make very clear the different usage between a rule and a rule factory.

### Updated sections

We now use a new folder and file to make a distinction between the permissions plugin and the custom rule.
Screenshot 2023-11-24 at 12 40 33

Updated naming make it clear when we are using the rule (used in catalog plugin) versus the factory (used in permissions plugin).
Screenshot 2023-11-24 at 12 40 47

Screenshot 2023-11-24 at 12 42 15

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — APPROVED** [review]
Thanks for the contribution @stanley-nicholl, this does make it easier to understand this section! 🚀 

Now we just need one of the maintainers to take a look

File: docs/permissions/custom-rules.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ For some use cases, you may want to define custom [rules](./concepts.md#resource
 
 ## Define a custom rule
 
-Plugins should export a rule factory that provides type-safety that ensures compatibility with the plugin's backend. The catalog plugin exports `createCatalogPermissionRule` from `@backstage/plugin-catalog-backend/alpha` for this purpose. Note: the `/alpha` path segment is temporary until this API is marked as stable. For this example, we'll define the rule in `packages/backend/src/plugins/permission.ts`, but you can put it anywhere that's accessible by your `backend` package.
+Plugins should export a rule factory that provides type-safety that ensures compatibility with the plugin's backend. The catalog plugin exports `createCatalogPermissionRule` from `@backstage/plugin-catalog-backend/alpha` for this purpose. Note: the `/alpha` path segment is temporary until this API is marked as stable. For this example, we'll define the rule in `packages/backend/src/customPermissionRules/isInSystem.ts`, but you can put it anywhere that's accessible by your `backend` package.
```
[the remark this answers is not in the document]
Comment (stanley-nicholl) [reply docs/permissions/custom-rules.md:11]: @tudi2d, sorry, might not have been clear here. Definitely not recommending the removal of the `src/plugins` folder, that would remain with all relevant plugin files in it (e.g. `permission.ts`, `auth.ts`, `catalog.ts`, etc.). But for custom rules, created a folder to store them that is outside of the `plugins` directory.

Custom permissions could certainly live inside the `permissions.ts` file, however, for this example, I found it clearer to explicitly import them (i.e. the rule and the factory) so there wasn't any confusion.

Another pattern could be to add a `permissions` folder to `src/plugins` which would contain `permission.ts` along with any custom permission files (e.g. `isInSystem.ts`).

Thoughts?

[the remark this answers is not in the document]
Comment (tudi2d) [reply docs/permissions/custom-rules.md:11]: thanks for clarifying - sounds all good to me. think people will get that they could rename it, just want to make sure you intend to move this out of `/plugins/permissions.ts` :)

**Review by tudi2d — APPROVED** [review]
let's get this in :shipit:

**Comment by stanley-nicholl** [comment]
Thanks, @tudi2d. @Rugvip, would you mind giving this a review for me to enable merging it in? It's a docs only change related to custom permissions.

**Review by freben — APPROVED** [review]
Sorry!

**Pull request merged**